Library/Homebrew/utils/bottles.rb

```diff
@@ -40,32 +40,40 @@ def extname_tag_rebuild(filename)
         HOMEBREW_BOTTLES_EXTNAME_REGEX.match(filename).to_a
       end
 
+      def bottle_file_list(bottle_file)
+        @bottle_file_list ||= {}
+        @bottle_file_list[bottle_file] ||= Utils.popen_read("tar", "-tzf", bottle_file)
```

Comment:
I don't quite follow why you want a hash here, but there's no need to explain if you don't have the time.

Reply:
Caching probably

Reply:
Yup, this is caching the `tar -tzf` shell output. Can't have a global cache because the results will differ based on bottle files. A `Hash` provides a cache keyed on the filename with speedy retrieval.

Reply:
Aha, yes, I see it now. Thanks. Neat design pattern.